src/test/java/com/team404/kainosproject/unittests/service/BandServiceTest.java

```diff
@@ -0,0 +1,168 @@
+    when(mockIndicator1.getSubCompetency()).thenReturn(mockSubCompetency);
+
+    when(mockIndicator2.getDescription()).thenReturn("test indicator two");
+    when(mockIndicator2.getBand()).thenReturn(mockBand2);
+    when(mockIndicator2.getSubCompetency()).thenReturn(mockSubCompetency);
+
+    // Mock a sub-competency
+    when(mockSubCompetency.getName()).thenReturn("test sub competency");
+    when(mockSubCompetency.getCompetency()).thenReturn(mockCompetency);
+
+    // Mock an indicator
+    when(mockCompetency.getName()).thenReturn("test competency");
+
+    // Mock a repository which will return the mock bands
+    when(bandRepository.findAll()).thenReturn(new ArrayList<>(){{
+      add(mockBand1);
+      add(mockBand2);
+    }});
+
+  }
+
+  @Test
+  public void when_getAllBandDtos_expect_correctDTOsAreCreated(){
+    setupMocks();
+    BandService testService = new BandService(bandRepository);
+    List<BandDto> result = (ArrayList<BandDto>) testService.getAllBandsDtos();
+    checkBandDtoStructure(result);
+  }
+
+
+  private void checkBandDtoStructure(List<BandDto> bands) {
+
+    assertAll(
+        () -> assertEquals("First test band had an unexpected name",
+            "test band one", bands.get(0).getBand()
+        ),
+        () -> assertEquals("Second test band had an unexpected name",
+            "test band two", bands.get(1).getBand()
+        ),
+
+        () -> assertEquals("First test band had an unexpected competency name",
+            "test competency", bands.get(0).getCompetencies().get(0).getName()
+        ),
+        () -> assertEquals("Second test band had an unexpected competency name",
+            "test competency", bands.get(1).getCompetencies().get(0).getName()
+        ),
+
+        () -> assertEquals("First test band had an unexpected sub-competency name",
+            "test sub competency", bands.get(0).getCompetencies().get(0).getIndicators().get(0).getName()
+        ),
+        () -> assertEquals("Second test band had an unexpected sub-competency name",
+            "test sub competency", bands.get(1).getCompetencies().get(0).getIndicators().get(0).getName()
+        ),
+
+        () -> assertEquals("First test band had an unexpected indicator description",
+            "test indicator one", bands.get(0).getCompetencies().get(0).getIndicators().get(0).getDescription()
+        ),
+        () -> assertEquals("Second test band had an unexpected indicator description",
+            "test indicator two", bands.get(1).getCompetencies().get(0).getIndicators().get(0).getDescription()
+        )
+    );
+
+    /* Expected is
+      [
+        {
+            'band' : 'test band one',
+            'competencies' : [
+                {
+                    'name' : 'test competency',
+                    'indicators' :
+                    [
+                        {'name' : 'test sub competency', 'description' : "test indicator one"}, {'name' : 'test sub competency', 'description' : "test indicator two"},
```

Comment:
Getting rid of this comment would be a good practice here, as this is a JSON result from api endpoint, easy to check